### PR TITLE
Support Link signup in saved payment mode

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- Modifies the Link inline signup view to have two modes, one of which supports the progressive disclosure without the checkbox.
- We now show Link sign up even when saving a card

### Demo (shows tapping privacy links before finishing payment)

https://github.com/stripe/stripe-ios/assets/88012362/1a0c0821-48d9-4693-b840-edc0fe7ed44a

## Motivation
- https://jira.corp.stripe.com/browse/MOBILESDK-1364

## Testing
- Manual
- New UI test

## Changelog
N/A

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift [hunk 1]
```diff
@@ -42,11 +42,8 @@ class PaymentSheetFormFactory {
     let cardBrandChoiceEligible: Bool
 
     var canSaveToLink: Bool {
-        // For Link private beta, only save cards in ".userSelectable" mode.
-        // We don't want to override the merchant's own "Save this card" checkbox.
         return (supportsLinkCard &&
                 paymentMethod == .stripe(.card) &&
-                saveMode != .userSelectable &&
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift:49]: Double check me here, but I think this new signup removes the userSelectable restriction

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+Card.swift [hunk 2]
```diff
@@ -20,7 +20,7 @@ extension PaymentSheetFormFactory {
                 merchantDisplayName: configuration.merchantDisplayName
             )
         )
-        let shouldDisplaySaveCheckbox: Bool = saveMode == .userSelectable && !canSaveToLink
+        let shouldDisplaySaveCheckbox: Bool = saveMode == .userSelectable
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+Card.swift:23]: Same here

**Comment by davidme-stripe** [comment]
Looks great to me! One more thing: Could we gate this based on the client-side `allowLinkV2Features` flag to make sure users don't see this until we ungate it? Should just be a matter of adding `allowLinkV2Features`, I added comments above.

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift (hunk 1 above)
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift:49]: Maybe: `(configuration.allowLinkV2Features || saveMode != .userSelectable) &&`

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+Card.swift (hunk 2 above)
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+Card.swift:23]: Maybe: ` && (configuration.allowLinkV2Features || !canSaveToLink)`

**Comment by porter-stripe** [comment]
@davidme-stripe Sounds good! Also added this one check: https://github.com/stripe/stripe-ios/pull/3166/commits/4bd84f8d72502c8bb52c3a02268ac63f2167f881#diff-644756fdfa33a2f4007e79278ae5ff1c4031244ca6e8f971c01da96f64a5062bR101

**Review by davidme-stripe — APPROVED** [review]

**Pull request merged**